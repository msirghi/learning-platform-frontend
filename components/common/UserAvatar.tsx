import React from 'react';

type Props = {
  img?: string;
};

const styles = { height: 35, width: 35, background: '#bbb', borderRadius: '50%' };

export const UserAvatar: React.FC<Props> = ({ img }) => {
  if (img) {
    return <img style={styles} src={img} />;
  }

  return <span style={styles} />;
};
