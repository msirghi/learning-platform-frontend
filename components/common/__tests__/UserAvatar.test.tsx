import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import { UserAvatar } from "../UserAvatar";

describe("UserAvatar component", () => {
  it("should match the snapshot", () => {
    const wrapper = shallow(<UserAvatar />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should return img on prop existance", () => {
    const wrapper = shallow(<UserAvatar img='text' />);
    const img = wrapper.find("img");
    expect(img).toHaveLength(1);
    expect(img.props().src).toBe("text");
  });
});
