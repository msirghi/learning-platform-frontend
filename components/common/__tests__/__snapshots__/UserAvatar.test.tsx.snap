// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserAvatar component should match the snapshot 1`] = `
<span
  style={
    Object {
      "background": "#bbb",
      "borderRadius": "50%",
      "height": 35,
      "width": 35,
    }
  }
/>
`;
